File: 8kyu/(8 kyu) Well of Ideas - Easy Version/(8 kyu) Well of Ideas - Easy Version.js
// 1 Plain Solution
function well(x) {
    let goodIdeas = 0;

    for (let i = 0; i < x.length; i++) {
        if (x[i] === 'good') {
            goodIdeas += 1;
        }
    }

    if (goodIdeas === 0) {
        return 'Fail!';
    } else if (goodIdeas > 0 && goodIdeas < 3) {
        return 'Publish!';
    } else {
        return 'I smell a series!';
    }
}
// 2 Optimized solution
function well(x) {
    let goodIdeas = 0;

    x.forEach(y => y === 'good' && goodIdeas++);

    if (!goodIdeas) return 'Fail!';
    if (goodIdeas === 1 || goodIdeas === 2) return 'Publish!';
    return 'I smell a series!';
}
// 3 Clever solution
function well(x) {
    const goodIdeas = x.filter(y => y === 'good').length;
    return (goodIdeas <= 2) ?
        ['Fail!', 'Publish!', 'Publish!'][goodIdeas]
        : 'I smell a series!';
}

File: 8kyu/(8 kyu) Well of Ideas - Easy Version/(8 kyu) Well of Ideas - Easy Version.rs
// 1 Plain Solution
fn well(x: &[&str]) -> &'static str {
    let mut goodIdeas = 0;
    for s in x {
        match *s {
            "good" => goodIdeas += 1,
            _ => {}
        }
    }
    if goodIdeas == 0 {
        "Fail!"
    } else if goodIdeas > 0 && goodIdeas < 3 {
        "Publish!"
    } else {
        "I smell a series!"
    }
}

// 2 Optimized Solution
fn well(x: &[&str]) -> &'static str {
    let goodIdeas = x.iter().filter(|&&s| s == "good").count();
    match goodIdeas {
        0 => "Fail!",
        1..=2 => "Publish!",
        _ => "I smell a series!",
    }
}

// 3 Clever Solution
fn well(x: &[&str]) -> &'static str {
    match x.iter().filter(|&&s| s == "good").count() {
        0 => "Fail!",
        1..=2 => "Publish!",
        _ => "I smell a series!",
    }
}
